### PR TITLE
Added robohatmm1 board defs.

### PR DESCRIPTION
**Pull request by wallarug**
As discussed in a prior issue https://github.com/adafruit/circuitpython/issues/1776, @tannewt  requested I add the board definition files for this board into the main repository.

The Robo HAT MM1 board will be listed on [CrowdSupply](https://www.crowdsupply.com/robotics-masters/robo-hat-mm1) soon and runs CircuitPython by Adafruit.

Build by:

```
make BOARD=robohatmm1
```

### CODE REVIEW

File: ports/atmel-samd/boards/robohatmm1/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,34 @@
+LD_FILE = boards/samd21x18-bootloader-external-flash-crystalless.ld
+USB_VID = 0x0005
```
Comment (tannewt) [inline ports/atmel-samd/boards/robohatmm1/mpconfigboard.mk]: Do you actually own this VID?

File: ports/atmel-samd/boards/robohatmm1/pins.c [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	{ MP_ROM_QSTR(MP_QSTR_RCH1), MP_ROM_PTR(&pin_PA07) },
+	{ MP_ROM_QSTR(MP_QSTR_RCH2), MP_ROM_PTR(&pin_PA06) },
+	{ MP_ROM_QSTR(MP_QSTR_RCH3), MP_ROM_PTR(&pin_PA05) },
+	{ MP_ROM_QSTR(MP_QSTR_RCH4), MP_ROM_PTR(&pin_PA04) },
+	
+	// Special Function (Primary)
+	{ MP_ROM_QSTR(MP_QSTR_I2C_SDA), MP_ROM_PTR(&pin_PA00) },
+	{ MP_ROM_QSTR(MP_QSTR_I2C_SCL), MP_ROM_PTR(&pin_PA01) },
+	
+	{ MP_ROM_QSTR(MP_QSTR_BATGUAGE), MP_ROM_PTR(&pin_PA02) },
```
Comment (tannewt) [inline ports/atmel-samd/boards/robohatmm1/pins.c]: IIRC we've standardized on `BATTERY` for this name.

File: ports/atmel-samd/boards/robohatmm1/mpconfigboard.mk (hunk 1 above)
Comment (wallarug) [reply ports/atmel-samd/boards/robohatmm1/mpconfigboard.mk]: No.  This is a placeholder until we get a new VID/PID for each product.  What should it be set to if we do not have a VID yet?

I'm hoping to have a real one before the end of next week.

File: ports/atmel-samd/boards/robohatmm1/pins.c (hunk 2 above)
Comment (wallarug) [reply ports/atmel-samd/boards/robohatmm1/pins.c]: No worries.  We can change that.

File: ports/atmel-samd/boards/robohatmm1/mpconfigboard.mk (hunk 1 above)
Comment (wallarug) [reply ports/atmel-samd/boards/robohatmm1/mpconfigboard.mk]: We have just put in a request for VID: 0x4D41 with a swag of PIDs.  I will update once confirmed.

File: ports/atmel-samd/boards/robohatmm1/pins.c (hunk 2 above)
Comment (tannewt) [reply ports/atmel-samd/boards/robohatmm1/pins.c]: Ok, cool. Let me know when I should review.

Comment (wallarug) [reply ports/atmel-samd/boards/robohatmm1/pins.c]: Should be all good now.  Could you please point me in the right direction for the .travis.yml file you mentioned?  I'm new at this 😄 .

Comment (sommersoft) [reply ports/atmel-samd/boards/robohatmm1/pins.c]: >  Could you please point me in the right direction for the .travis.yml

First board build line is here: https://github.com/adafruit/circuitpython/blob/master/.travis.yml#L24

Use one of the "TRAVIS_BOARDS" environment variable to add your board to.

Comment (wallarug) [reply ports/atmel-samd/boards/robohatmm1/pins.c]: Thank you!  All done.

**Comment by wallarug** [comment]
@tannewt  It seems all issues have now been resolved.

**Comment by wallarug** [comment]
Thank you @tannewt !  FYI the campaign is now up and running as well...

https://www.crowdsupply.com/robotics-masters/robo-hat-mm1/

**Pull request merged**